python.md

```diff
@@ -31,10 +31,74 @@ d = {
 }
 ```
 
-## Exception handling
+## [Declaring exceptions](https://stackoverflow.com/questions/1319615/proper-way-to-declare-custom-exceptions-in-modern-python/26938914#26938914)
+### The example
 
-* When raising exceptions, try to include enough information to help debugging: what was expected? What actually happened?
-* When writing controllers which handle a lot of Exceptions, try to encapsulate the logic within one `try..except`
+```
+class MyAppValueError(ValueError):
+    def __init__(self, foo, *args):
+        # Special attribute you desire with your Error,
+        # perhaps the value that caused the error?:
+        self.foo = foo
+
+        # Allow users initialize misc. arguments as any other builtin Error
+        super(MyAppValueError, self).__init__(message, foo, *args)
+```
+
+There's no need to write your own `__str__` or `__repr__`. The builtin ones
+are very nice.
+
+### The example, expanded
+
+**Name** your exceptions [in accordance with
+PEP8](https://www.python.org/dev/peps/pep-0008/#exception-names), i.e. use
+suffixes like `Error` or `Warning` where appropriate.
+
+Exception classes you declare **should inherit** (either directly or indirectly)
+from the `Exception` class.  If there's more that can be told about the nature
+of the exceptional situation, pick a more specific exception class to inherit
+from, from the hierarchy of [predefined exceptions](https://docs.python.org/2/library/exceptions.html#exception-hierarchy).
+
+The last parameter of `__init__()` method **should be `*args`** to allow for
+passing arbitrary number of arguments, which in turn are passed on to
+`super(...).__init__(..., *args)`.  That way, the Liskov Substitution
+Principle isn't violated between your exception class and the `Exception`
+class.
```

Comment:
@ldoran: Sorry for pushing after the PR has been accepted but I thought another section about `*args*` was needed.

Reply:
No worries, I looked at the update and I shippit.